README.md

```diff
@@ -1 +1,338 @@
 ![travis ci](https://travis-ci.org/electrode-io/react-native-electrode-bridge.svg?branch=master)
+
+
+Electrode React Native Bridge
+-----------------------------
+
+This project is essentially a react native library, consisting of a JavaScript module and an associated Android Native Module and iOS Native module.
+
+It is built on top of the react native built-in bridging constructs (to communicate between the react native JS side and Native side) and offers a clean bi-directional communication API, exposing methods to send events and/or requests from/to any side of the bridge (JS/Native).
+It offers more options and flexibility to communicate between the JS/Native side that is not offered out of the box by react native. Ultimately it can help with integrating react native applications into existing native code bases.
+It might be used as one of the basic building block of react native apps and native modules.
+
+Here is a non-exhaustive list of a few reasons to use this library as the low level communication bridge instead of the built-in react native constructs :
+
+- Isolates the host application from react native library types and specifics (vanilla Java/Android/Swift implementation)
+- Messages can be sent either to the other side of the bridge or on the same side (or both)
+- Request timeout supported
+- Offer an easy way to send requests from Native to JS side.
+- Supports strong types
+- More than a bridge, it can be used as a message hub allow react native apps / native modules intercommunication
+
+The bridge API is built around two messaging idioms, `events` and `requests` :
+
+
+##### Events
+- An event is a fire & forget message type.
+- You should emit an event whenever you are not expecting any kind of response back from whoever is listening for this specific event. Because of this, there can be multiple listeners for a given event name.
+- When an event is emitted from one side of the spectrum (Native or JS), the bridge sends notification to both native and react native bridge and all the registered listeners will be notified.
```

Comment:
For the first time user he/she might be confused between `native and react native bridge` unless we have defined it else where. I'd just keep it as bridge to avoid confusions..